src/apps/Calculators/IntervalPattern/IPModeReadoutTable.js

```diff
@@ -0,0 +1,63 @@
+import { handleSetPlayerClick } from "../../../eventHandlers";
+import addParenthesesToModeReadout from "../../../util/addParenthesesToModeReadout";
+import generateIPScl from "./util/generateIPScl";
+
+const IPModeReadoutTable = ({ index, ipObject, stateData, setStateData }) => {
+  const sclData = addParenthesesToModeReadout([...ipObject.pattern]);
```

Comment:
Spreading an array into a new array literal is circular - it's the same as just passing the argument:

```suggestion
  const sclData = addParenthesesToModeReadout(ipObject.pattern);
```